test/unit/stats.js

```diff
@@ -0,0 +1,793 @@
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is not in the database',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: 9999999
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is not found in the database');
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+  });
+
+  describe('createStat(req)', function() {
+    // Post a new row of stats into stats database
+    xit('should add new stats to the database', function(done) {
+      var req = {
+        body: {
+          student_id: 2,
+          event_id: 2,
+          height: 320,
+          weight: 54,
+          pacer: 382,
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStat(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.lengthOf(data, statCount + 1);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6, fakeStat7]);
+        done();
+      });
+    });
+
+    // TODO Insert test for trying to post if stat already exists
+    // (should do nothing. Make sure of that.)
+
+    // Attempt to post existing stat
+    xit('adding an existant set of stat should do nothing', function(done) {
+      var req = {
+        body: {
+          student_id: 1,
+          event_id: 1,
+          height: 5,
+          weight: 5,
+          pacer: 5
+        }
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
+           fakeStat4, fakeStat5, fakeStat6]);
+
+        return stats.createStat(req);
+      })
+      .then(function() {
+        return stats.getStats({});
+      })
+      .then(function(data) {
+        assert.lengthOf(data, statCount);
+        assert.deepEqual([fakeStat, fakeStat2, fakeStat3,
+          fakeStat4, fakeStat5, fakeStat6]);
+        assert.equal(result.message,
+        'Unable to add student because this set of stats already exists');
+        done();
+      });
+    });
+
+    // Attempt to post stats but missing body
+    xit('attempting to post stats with missing body will result in an error', function(done) {
+      var req = {
+        body: {}
+      };
+
+      var promise = stats.getStats({});
+      var statCount;
+
+      promise.then(function(data) {
+        statCount = data.length;
+        assert.deepEqual(data, [fakeStat, fakeStat2, fakeStat3,
```

Comment:
No need to check this on failure cases (before and after results don;t matter much). You don't have to delete it though